modules/secrets/text/README.md

```diff
@@ -9,22 +9,57 @@ This module handle the creation and managment of Text Secrets by leveraging the
     - v1.3.8
     - v1.3.9
     - v1.4.0
+    - v1.4.2
 
 ## Providers
+This module requires that the calling template has defined the [Harness Provider - Docs](https://registry.terraform.io/providers/harness/harness/latest/docs) authentication.
 
+### Example setup of the Harness Provider
+```
+# Provider Setup Details
+variable "harness_platform_url" {
+  type        = string
+  description = "[Optional] Enter the Harness Platform URL.  Defaults to Harness SaaS URL"
+  default     = "https://app.harness.io/gateway"
+}
+
+variable "harness_platform_account" {
+  type        = string
+  description = "[Required] Enter the Harness Platform Account Number"
+  default     = null # If Not passed, then the ENV HARNESS_ACCOUNT_ID will be used
+  sensitive   = true
+}
+
+variable "harness_platform_key" {
+  type        = string
+  description = "[Required] Enter the Harness Platform API Key for your account"
+  default     = null # If Not passed, then the ENV HARNESS_PLATFORM_API_KEY will be used
+  sensitive   = true
+}
+
+provider "harness" {
+  endpoint         = var.harness_platform_url
+  account_id       = var.harness_platform_account
+  platform_api_key = var.harness_platform_key
+}
+
+```
+
+### Terraform required providers declaration
```

Comment:
```suggestion
```

### Example setup of the Harness Provider with environment variables

You can also set up authentication with Harness through environment variables. To do this set the following items in your environment:

- HARNESS_ENDPOINT: Harness Platform URL, defaults to Harness SaaS URL: https://app.harness.io/gateway
- HARNESS_ACCOUNT_ID: Harness Platform Account Number
- HARNESS_PLATFORM_API_KEY: Harness Platform API Key for your account

### Terraform required providers declaration
```

how about something like this?